File: recursion.py
def f(arr):
    if not arr:
        return 0
    return arr[0] + f(arr[1:])

def e(arr):
    if not arr:
        return 0
    return 1 + e(arr[1:])    

def k(arr):
    if len(arr) == 1:
        return arr[0]
    sub_max = k(arr[1:])
    return arr[0] if arr[0] > sub_max else sub_max


data = [1, 3, 4, 5]


print(e(data))

print(f(data))
print(sum(data))
print(k(data))

File: recursion.js
'use strict';

const f = arr => {
  if (arr.length === 0) {
    return 0;
  }
  return arr[0] + f(arr.slice(1));
}

const e = arr => {
  if (arr.length === 0) {
    return 0;
  }
  return 1 + e(arr.slice(1));
}

const k = arr => {
  if (arr.length === 1) {
    return arr[0];
  }
  const subMax = k(arr.slice(1));
  return arr[0] > subMax ? arr[0] : subMax;

}

const data = [1, 2, 3, 4, 5];

console.log(f(data));
console.log(e(data));
console.log(k(data));